daffodil-codegen-c/src/main/scala/org/apache/daffodil/codegen/c/DaffodilCCodeGenerator.scala

```diff
@@ -275,6 +276,7 @@ object DaffodilCCodeGenerator
       case g: ElementParseAndUnspecifiedLength =>
         elementParseAndUnspecifiedLengthGenerateCode(g, cgState)
       case g: ElementUnused => noop(g)
+      case g: Gram if g eq EmptyGram => noop(g)
```

Comment:
It might better to add this as the first case instead:
```scala
case g: Gram if g.isEmpty => noop(g)
```
This will still noop `EmptyGram`'s like you have, but also noop non-`EmptyGram` grammars that determine that they are empty. I'm not sure if it's possible for codgen-c to walk into those or if you'll only ever see empty grammars that are `EmptyGram`, but it wouldn't hurt.

Reply:
Thanks, I have improved the fix as suggested and added a new test case with a test schema containing an empty gram.  Ready for review again.